diff --git a/node_modules/@cosmjs/cosmwasm-stargate/build/modules/wasm/queries.js b/node_modules/@cosmjs/cosmwasm-stargate/build/modules/wasm/queries.js
index 8f87522..f1bae9c 100644
--- a/node_modules/@cosmjs/cosmwasm-stargate/build/modules/wasm/queries.js
+++ b/node_modules/@cosmjs/cosmwasm-stargate/build/modules/wasm/queries.js
@@ -55,7 +55,7 @@ function setupWasmExtension(base) {
                 return queryService.RawContractState(request);
             },
             queryContractSmart: async (address, query) => {
-                const request = { address: address, queryData: (0, encoding_1.toAscii)(JSON.stringify(query)) };
+                const request = { address: address, queryData: (0, encoding_1.toUtf8)(JSON.stringify(query)) };
                 const { data } = await queryService.SmartContractState(request);
                 // By convention, smart queries must return a valid JSON document (see https://github.com/CosmWasm/cosmwasm/issues/144)
                 let responseText;
